### ISSUE TITLE
Server appears in network neighborhood, but cannot connect (unless it is an IP address)

### ISSUE THREAD

**Issue by octogonz** [christgau/wsdd#104]
If I invoke this command:
```shell
$ wsdd.py  -i eth1 -n SERVER -w WORKGROUP -4
```
...then the "SERVER" computer appears under the "Network" view in Windows Explorer.  But when I try to browse its shares, Windows says it cannot access the share. 👎

Whereas if I invoke this command:
```shell
$ wsdd.py  -i eth2 -n 192.168.3.201 -w WORKGROUP -4
```
...then the raw IP address is shown in Windows Explorer, and everything works fine. 👍

Without running `wsdd.py`, I can also manually enter `\\192.168.3.201\MyShare` in the Windows Explorer address bar, and that works as well.  (And obviously if I add `SERVER` to my `drivers/etc/hosts` file on the Windows computer, that works too.)

It seems that `wsdd.py` is not communicating correct IP address to Windows.  (?)

How to diagnose this problem?  The Samba logs don't show no activity even with `log level = 3`, as if the client never tried to connect.

I also tried using the `-W` option for `wsdd.py` which dumps the SOAP packets.  And I do see `<wsd:XAddrs>http://192.168.3.201:3702/2f838418-e833-46ef-997a-613d1b515cd9</wsd:XAddrs>` but that is referring to the port used by `wsdd.py`.  I can't figure out where in the protocol it tells Windows how to find Samba.  (?)

**Comment by christgau** [comment]
> If I invoke this command: ...then the "SERVER" computer appears under the "Network" view in Windows Explorer. 

That's everything wsdd should do...

> But when I try to browse its shares, Windows says it cannot access the share. 👎

... and this is out of wsdd's scope.

> Whereas if I invoke this command ...then the raw IP address is shown in Windows Explorer, and everything works fine. 👍
> Without running `wsdd.py`, I can also manually enter `\\192.168.3.201\MyShare` in the Windows Explorer address bar, and that works as well. (And obviously if I add `SERVER` to my `drivers/etc/hosts` file on the Windows computer, that works too.)

You already pointed out the main issue: You need a working name resolution (see https://github.com/christgau/wsdd/issues/57#issuecomment-689730116). This is apparently not the case. Therefore you also see no traces in the Samba log. 

> It seems that `wsdd.py` is not communicating correct IP address to Windows. (?)

Wsdd does not communicate any IP address to Windows that is related to Samba. It just says 'Hi, I'm a host and if we talk to each other a little more, I will tell you that I am a "Computer" device in the network and my (resolvable) name is [whatever wsdd was configured for]'

> I also tried using the `-W` option for `wsdd.py` which dumps the SOAP packets. And I do see `<wsd:XAddrs>http://192.168.3.201:3702/2f838418-e833-46ef-997a-613d1b515cd9</wsd:XAddrs>` but that is referring to the port used by `wsdd.py`. I can't figure out where in the protocol it tells Windows how to find Samba. (?)

The XAddr(s) are used to exchange metadata about the device. When Windows find our that the host is a 'computer' by contacting the XAddr(s), it implies the host offers SMB, presents it in the network view and if you double click, it tries to resolve the name and connects to the SMB port.

**Comment by feeRnt** [comment]
For anyone else coming across this, here's a simple bash function to make your Samba host auto-discoverable and accessible from a Windows machine using Network Sharing, based on octogonz' comment.

 The regex using sed could be improved, but it worked for me on my IPv4 LAN. Unfortunately you will not get the hostname of the machine running the Samba service, but at least the exposed Samba shares will be navigable from Windows File explorer without having to manually type it in yourself.

The key is in the `ip route` command, and its "dev" and "src" fields from the output, which show your interface (-i) and hostnames (-n) respectively that you can use. 
You can modify and change your workgroup (-w) or IPv4/6 selection with the -4 or -6 switches like shown above. 

```bash
samba_discover_wsdd() {
	if [ "$(systemctl is-active smb.service)" == "inactive" ]; then
		echo "Start Samba first! \"\$sudo samba\""
	else
		__wsdd_host_dev="$(ip route show | sed -nE 's|^default.*dev (.*) proto.*|\1|p')" 
		__wsdd_host_ip="$(ip route show | sed -nE 's|^.* src (.*) metric .*|\1|p')"

		wsdd -i "$__wsdd_host_dev" -n "$__wsdd_host_ip" & echo -e "Web Services Discovery Daemon up and running with \n Interface = "$__wsdd_host_dev" \n IP = "$__wsdd_host_ip""
		unset __wsdd_host_dev
		unset __wsdd_host_ip
	fi
}
```

**Comment by feeRnt** [comment]
Ah.. I was suspecting that could be an issue. For me, I didn't get any src IP in my "default" route, e.g.:
```
$ ip route show
default via 192.168.0.1 dev wlan0 proto static metric 600 
192.168.0.0/24 dev wlan0 proto kernel scope link src 192.168.0.107 metric 600 
```

In that case we could adjust the script a little:
	`__wsdd_host_ip="$(ip route show | sed -nE 's|^.* src (.*) metric .*|\1|p')" | head -1`

This should output only the *first* src IP in the list, which should be for the LAN route with the highest priority, or the default.

I tested the script with your output and it seemed to output the IP properly.
I updated the above script with this change, hopefully it works.

I think there's a simple way to do this without using head, using the methods described [here](https://www.gnu.org/software/sed/manual/html_node/Branching-and-flow-control.html) or [here](https://superuser.com/questions/1555147/how-to-exit-sed-after-the-first-match), but I'm not sure how, so this is the simplest to me now.
